File: day1/part1.py
import wafle as wf

parsed = wf.M(filter(bool, open("inp.txt", "r").readlines())) | str.strip | str.split | (lambda k: (int(k[0]), int(k[1])))

a, b = (zip(*parsed) >= wf.mapper) | list
a.sort()
b.sort()

_ = wf.M(zip(a, b))
_ |= lambda k: k[0] - k[1]
_ |= abs

print(_ >= sum)

File: haskell/day1/part1.hs
import System.IO
import Data.List (sort)

isSep :: Char -> Bool
isSep '\n' = True
isSep ' ' = True
isSep c = False

wordss :: String -> [String]
wordss s = case dropWhile isSep s of
    "" -> []
    s' -> w : wordss s'' where (w, s'') = break isSep s'

main :: IO ()
main = do
    h <- openFile "inp.txt" ReadMode
    c <- hGetContents h
    let k = zip (map (read::(String -> Int)) (wordss c)) [1..]
    let (a, b) = ([ fst n | n <- k, odd $ snd n], [ fst n | n <- k, even $ snd n])
    let c = zipWith (curry (\ n -> abs (fst n - snd n))) (sort a) (sort b)
    print $ sum c